flyteplugins/go/tasks/logs/logging_utils_test.go

```diff
@@ -566,3 +566,204 @@ func TestGetLogsForContainerInPod_Flyteinteractive(t *testing.T) {
+						},
+					},
+				},
+			},
+			&core.TaskTemplate{
+				Config: map[string]string{
+					"link_type":                 "vscode",
+					"port":                      "65535",
+					"route":                     "a-route",
+					"generic_dynamic_log_links": `{"links": [{"name": "vscode", "display_name": "Other logs", "template_uri": "coming://from/other/place"}]}`,
+				},
+			},
+			[]*core.TaskLog{
+				{
+					Uri:           "https://abc.com:65535/a-route",
+					MessageFormat: core.TaskLog_JSON,
+					Name:          "vscode link my-Suffix",
+				},
+				{
+					Uri:           "coming://from/other/place",
+					MessageFormat: core.TaskLog_JSON,
+					Name:          "Other logs",
+				},
+			},
+		},
+		{
+			"Generic Dynamic Log Links disabled and K8s enabled and generic dynamic log links config present in TaskTemplate",
+			&LogConfig{
+				IsKubernetesEnabled:           true,
+				KubernetesTemplateURI:         "https://k8s.com/{{ .namespace }}/{{ .podName }}/{{ .containerName }}/{{ .containerId }}",
+				GenericDynamicLogLinksEnabled: false,
+			},
+			&core.TaskTemplate{
+				Config: map[string]string{
+					"link_type":                 "abc",
+					"generic_dynamic_log_links": `{"links": [{"display_name": "Other logs", "template_uri": "coming://from/other/place"}]}`,
+				},
+			},
+			[]*core.TaskLog{
+				{
+					Uri:           "https://k8s.com/my-namespace/my-pod/ContainerName/ContainerID",
+					MessageFormat: core.TaskLog_JSON,
+					Name:          "Kubernetes Logs my-Suffix",
+				},
+			},
+		},
+		{
+			"Generic Dynamic Log Links enabled and K8s enabled and generic dynamic log links config present in TaskTemplate",
+			&LogConfig{
+				IsKubernetesEnabled:           true,
+				KubernetesTemplateURI:         "https://k8s.com/{{ .namespace }}/{{ .podName }}/{{ .containerName }}/{{ .containerId }}",
+				GenericDynamicLogLinksEnabled: true,
+			},
+			&core.TaskTemplate{
+				Config: map[string]string{
+					"link_type":                 "abc",
+					"generic_dynamic_log_links": `{"links": [{"name": "abc", "display_name": "Other logs", "template_uri": "coming://from/other/place"}]}`,
+				},
+			},
+			[]*core.TaskLog{
+				{
+					Uri:           "https://k8s.com/my-namespace/my-pod/ContainerName/ContainerID",
+					MessageFormat: core.TaskLog_JSON,
+					Name:          "Kubernetes Logs my-Suffix",
+				},
+				{
+					Uri:           "coming://from/other/place",
+					MessageFormat: core.TaskLog_JSON,
+					Name:          "Other logs",
+				},
+			},
+		},
+		{
+			"Multiple Generic Dynamic Log Links defined in task template",
+			&LogConfig{
+				GenericDynamicLogLinksEnabled: true,
+			},
+			&core.TaskTemplate{
+				Config: map[string]string{
+					"link_type":                 "abc,defg",
+					"generic_dynamic_log_links": `{"links": [{"name": "abc", "display_name": "Other logs", "template_uri": "coming://from/abc"},{"name": "defg", "display_name": "defg logs", "template_uri": "coming://from/defg"}]}`,
```

Comment:
<div>


<div id="suggestion">
<div id="issue"><b>Consider splitting long configuration line</b></div>
<div id="fix">

Consider splitting the long line containing `generic_dynamic_log_links` configuration into multiple lines for better readability. The current line exceeds 120 characters which may impact code maintainability.
</div>
<details>
<summary>
<b>Code suggestion</b>
</summary>
<blockquote>Check the AI-generated fix before applying</blockquote>
<div id="code">


```suggestion
					"generic_dynamic_log_links": `{
						"links": [{"name": "abc", "display_name": "Other logs", "template_uri": "coming://from/abc"},
						          {"name": "defg", "display_name": "defg logs", "template_uri": "coming://from/defg"}]
					}`,
```

</div>
</details>
</div>


<small><i>Code Review Run <a href=https://github.com/flyteorg/flyte/pull/5066#issuecomment-2563965363>#51871a</a></i></small>
</div>

---
Is this a valid issue, or was it incorrectly flagged by the Agent?
- [ ] it was incorrectly flagged